src/librustc_back/target/mod.rs

```diff
@@ -287,6 +287,15 @@ impl Target {
                         )
                     );
             } );
+            ($key_name:ident, optional) => ( {
+                let name = (stringify!($key_name)).replace("_", "-");
+                obj.find(&name[..])
+                    .map(|o| {
+                        base.options.$key_name = o
+                            .as_string()
+                            .map(|s| s.to_string() )
+                    });
```

Comment:
This may be a little clearer as:

```
if let Some(o) = obj.find(name) {
    base.options.$key_name = o.as_string().map(|s| s.to_string());
}
```